tools/ci-operator-prowgen/main_test.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	ciop "github.com/openshift/ci-operator/pkg/api"
+)
+
+func checkCondition(t *testing.T, forHumans string, condition bool) {
+	if !condition {
+		t.Errorf("'%s' does not hold", forHumans)
+	}
+}
+
+func checkString(t *testing.T, member, value, expected string) {
+	if value != expected {
+		t.Errorf(
+			"%s not set correctly: expected '%s', got '%s'",
+			member,
+			expected,
+			value,
+		)
+	}
+}
+
+func TestGeneratePodSpec(t *testing.T) {
+	podSpec := generatePodSpec("organization", "repo", "branch", "target")
+	checkString(t, "podSpec.ServiceAccountName", podSpec.ServiceAccountName, "ci-operator")
+	checkString(t, "podSpec.Containers[0].Image", podSpec.Containers[0].Image, "ci-operator:latest")
+	checkString(t, "podSpec.Containers[0].Command[0]", podSpec.Containers[0].Command[0], "ci-operator")
+	checkString(t,
+		"podSpec.Containers[0].Args",
+		strings.Join(podSpec.Containers[0].Args, " "),
+		"--artifact-dir=$(ARTIFACTS) --target=target",
+	)
+	checkString(t, "podSpec.Containers[0].Env[0].Name", podSpec.Containers[0].Env[0].Name, "CONFIG_SPEC")
+	checkString(t,
+		"podSpec.Containers[0].Env[0].ValueFrom.ConfigMapKeyRef.LocalObjectReference.Name",
+		podSpec.Containers[0].Env[0].ValueFrom.ConfigMapKeyRef.LocalObjectReference.Name,
+		"ci-operator-organization-repo",
+	)
+	checkString(t,
+		"podSpec.Containers[0].Env[0].ValueFrom.ConfigMapKeyRef.Key",
+		podSpec.Containers[0].Env[0].ValueFrom.ConfigMapKeyRef.Key,
+		"branch.json",
+	)
+}
+
+func TestGeneratePodSpecWithAdditionalArgs(t *testing.T) {
```

Comment:
Idiom in Go is to use a `for` loop and sub-tests for similar tests like this https://blog.golang.org/subtests

Reply:
OK